File: TicTacToe.py
# Tic-Tac-Toe Game

# Constants for players
PLAYER_X = "X"
PLAYER_O = "O"
EMPTY = " "
board = [EMPTY for _ in range(9)]

def display_board(board):
    print(f"{board[0]} | {board[1]} | {board[2]}")
    print("--|---|--")
    print(f"{board[3]} | {board[4]} | {board[5]}")
    print("--|---|--")
    print(f"{board[6]} | {board[7]} | {board[8]}")

def is_game_over(board):
    winning_positions = [
        # Horizontal
        (0, 1, 2), (3, 4, 5), (6, 7, 8),
        # Vertical
        (0, 3, 6), (1, 4, 7), (2, 5, 8),
        # Diagonal
        (0, 4, 8), (2, 4, 6)
    ]
    
    for a, b, c in winning_positions:
        if board[a] == board[b] == board[c] != EMPTY:
            return True
            
    return EMPTY not in board  # Check for tie if no winner

def get_player_input(player):
    while True:
        position = input(f"Player {player}, enter your position (1-9): ")
        if position.isdigit() and 1 <= int(position) <= 9:
            index = int(position) - 1
            if board[index] == EMPTY:
                return index
            else:
                print("Position already taken. Please try again.")
        else:
            print("Invalid input. Please try again.")

def play_game():
    current_player = PLAYER_X
    while not is_game_over(board):
        display_board(board)
        position = get_player_input(current_player)
        board[position] = current_player
        current_player = PLAYER_O if current_player == PLAYER_X else PLAYER_X

    display_board(board)
    if EMPTY not in board:
        print("It's a tie!")
    else:
        winner = "Player X" if current_player == PLAYER_O else "Player O"
        print(f"{winner} wins!")

if __name__ == "__main__":
    play_game()

File: TicTacToe.cs
using System;

class TicTacToe
{
    static char[] board = new char[9] { ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ' };

    static void DisplayBoard()
    {
        Console.WriteLine($"{board[0]} | {board[1]} | {board[2]}");
        Console.WriteLine("--|---|--");
        Console.WriteLine($"{board[3]} | {board[4]} | {board[5]}");
        Console.WriteLine("--|---|--");
        Console.WriteLine($"{board[6]} | {board[7]} | {board[8]}");
    }

    static bool IsGameOver()
    {
        for (int i = 0; i < 9; i += 3)
        {
            if (board[i] == board[i + 1] && board[i + 1] == board[i + 2] && board[i] != ' ')
                return true;
        }

        for (int i = 0; i < 3; i++)
        {
            if (board[i] == board[i + 3] && board[i + 3] == board[i + 6] && board[i] != ' ')
                return true;
        }

        if (board[0] == board[4] && board[4] == board[8] && board[0] != ' ')
            return true;
        if (board[2] == board[4] && board[4] == board[6] && board[2] != ' ')
            return true;

        if (Array.IndexOf(board, ' ') == -1)
            return true;

        return false;
    }

    static int GetPlayerInput(char player)
    {
        while (true)
        {
            Console.Write($"Player {player}, enter your position (1-9): ");
            string input = Console.ReadLine();
            if (int.TryParse(input, out int position) && position >= 1 && position <= 9 && board[position - 1] == ' ')
                return position - 1;
            else
                Console.WriteLine("Invalid input. Please try again.");
        }
    }

    static void Main()
    {
        char currentPlayer = 'X';
        while (!IsGameOver())
        {
            DisplayBoard();
            int position = GetPlayerInput(currentPlayer);
            board[position] = currentPlayer;
            currentPlayer = currentPlayer == 'X' ? 'O' : 'X';
        }

        DisplayBoard();
        if (Array.IndexOf(board, ' ') == -1)
            Console.WriteLine("It's a tie!");
        else
            Console.WriteLine($"Player {(currentPlayer == 'X' ? 'O' : 'X')} wins!");
    }
}